packages/headless/src/api/service/case-assist/case-assist-params.ts

```diff
@@ -0,0 +1,85 @@
+import {HTTPContentType, HttpMethods} from '../../platform-client';
+import {BaseParam} from '../../platform-service-params';
+
+export interface CaseAssistIdParam {
+  caseAssistId: string;
+}
+
+export interface FieldsParam {
+  fields: Record<string, string>;
+}
+
+export type CaseAssistParam = BaseParam & CaseAssistIdParam;
+
+/**
+ * Builds a base request for calling the Case Assist API.
+ *
+ * @param req - The Case Assist request parameters.
+ * @param method - The HTTP method used to issue the request.
+ * @param contentType - The request content type.
+ * @param path - The request path, relative to the Case Assist path.
+ * @param queryStringArguments - The arguments to pass in the query string.
+ * @returns The built request information.
+ */
+export const baseCaseAssistRequest = (
+  req: CaseAssistParam,
+  method: HttpMethods,
+  contentType: HTTPContentType,
+  path: string,
+  queryStringArguments: Record<string, string> = {}
+) => {
+  validateCaseAssistRequestParams(req);
+
+  const baseUrl = `${req.url}/rest/organizations/${req.organizationId}/caseassists/${req.caseAssistId}${path}`;
+  const queryString = buildQueryString(queryStringArguments);
+  const effectiveUrl = queryString ? `${baseUrl}?${queryString}` : baseUrl;
+
+  return {
+    accessToken: req.accessToken,
+    method,
+    contentType,
+    url: effectiveUrl,
+  };
+};
+
+export const prepareSuggestionRequestFields = (
+  fields: Record<string, string>
+) =>
+  Object.keys(fields)
+    .filter((fieldName) => fields[fieldName] !== '')
+    .reduce(
+      (result, fieldName) => ({
+        ...result,
+        [fieldName]: {
+          value: fields[fieldName],
+        },
+      }),
+      {} as Record<string, unknown>
+    );
+
+const validateCaseAssistRequestParams = (req: CaseAssistParam) => {
```

Comment:
As I said above, it's okay to do those validation + error throw here, but the best place to do it would be at the "entry point" for those param. 

They must all come from some redux action somewhere upstream, and if we can catch the invalidity closer to the entry point, it's 100% easier to debug from an end user perspective.

Versus getting an error here will be very cryptic for most users and will be harder for them to debug the source of the problem.

So, just a mental note for you I guess ! ;)